crates/sui-rest-api/src/objects.rs

```diff
@@ -285,7 +285,7 @@ impl From<(DynamicFieldKey, DynamicFieldIndexInfo)> for DynamicFieldInfo {
             parent: parent.into(),
             field_id: field_id.into(),
             dynamic_field_type: dynamic_field_type.into(),
-            name_type: type_tag_core_to_sdk(name_type),
+            name_type: type_tag_core_to_sdk(name_type).expect("object types must be valid"),
```

Comment:
My personal taste would be to make this fallible as well, because it doesn't seem impossible that someone might feed a bad `DynamcFieldIndexInfo` in here.